monitor.go

```diff
@@ -128,6 +133,94 @@ func Monitor(executable, port string, options *compileopts.Options) error {
 	return <-errCh
 }
 
+// SerialPortInfo is a structure that holds information about the port and its
+// associated TargetSpec.
+type SerialPortInfo struct {
+	Name   string
+	IsUSB  bool
+	VID    string
+	PID    string
+	Target string
+	Spec   *compileopts.TargetSpec
+}
+
+// ListSerialPort returns serial port information and any detected TinyGo
+// target
+func ListSerialPorts() ([]SerialPortInfo, error) {
+	maps, err := GetTargetSpecs()
+	if err != nil {
+		return nil, err
+	}
+
+	portsList, err := enumerator.GetDetailedPortsList()
+	if err != nil {
+		return nil, err
+	}
+
+	serialPortInfo := []SerialPortInfo{}
+	for _, p := range portsList {
+		info := SerialPortInfo{
+			Name:  p.Name,
+			IsUSB: p.IsUSB,
+			VID:   p.VID,
+			PID:   p.PID,
+		}
+		vid := strings.ToLower(p.VID)
+		pid := strings.ToLower(p.PID)
+		for k, v := range maps {
+			usbInterfaces := v.SerialPort
+			for _, s := range usbInterfaces {
+				parts := strings.Split(s, ":")
+				if len(parts) != 2 {
+					continue
+				}
+				if vid == strings.ToLower(parts[0]) && pid == strings.ToLower(parts[1]) {
+					info.Target = k
+					info.Spec = v
+				}
+			}
+		}
+		serialPortInfo = append(serialPortInfo, info)
+	}
+
+	return serialPortInfo, nil
+}
+
+func GetTargetSpecs() (map[string]*compileopts.TargetSpec, error) {
```

Comment:
This file `monitor.go` seems like an odd place to move this function.

Reply:
`GetTargetSpecs() has been moved to the compileopts package.